java/vector/src/test/java/org/apache/arrow/vector/TestListViewVector.java

```diff
@@ -0,0 +1,605 @@
+
+      /* get listViewVector value at index 1 -- the value itself is a listViewVector */
+      result = listViewVector.getObject(1);
+      resultSet = (ArrayList<ArrayList<Long>>) result;
+
+      assertEquals(3, resultSet.size()); /* 3 inner lists at index 1 */
+      assertEquals(1, resultSet.get(0).size()); /* size of the first inner list */
+      assertEquals(2, resultSet.get(1).size()); /* size of the second inner list */
+      assertEquals(3, resultSet.get(2).size()); /* size of the third inner list */
+
+      list = resultSet.get(0);
+      assertEquals(Long.valueOf(10), list.get(0));
+
+      list = resultSet.get(1);
+      assertEquals(Long.valueOf(15), list.get(0));
+      assertEquals(Long.valueOf(20), list.get(1));
+
+      list = resultSet.get(2);
+      assertEquals(Long.valueOf(25), list.get(0));
+      assertEquals(Long.valueOf(30), list.get(1));
+      assertEquals(Long.valueOf(35), list.get(2));
+
+      /* check underlying bitVector */
+      assertFalse(listViewVector.isNull(0));
+      assertFalse(listViewVector.isNull(1));
+
+      final ArrowBuf offSetBuffer = listViewVector.getOffsetBuffer();
+      final ArrowBuf sizeBuffer = listViewVector.getSizeBuffer();
+
+      // check offset buffer
+      assertEquals(0, offSetBuffer.getInt(0 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+      assertEquals(2, offSetBuffer.getInt(1 * BaseRepeatedValueViewVector.OFFSET_WIDTH));
+
+      // check size buffer
+      assertEquals(2, sizeBuffer.getInt(0 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+      assertEquals(3, sizeBuffer.getInt(1 * BaseRepeatedValueViewVector.SIZE_WIDTH));
+    }
+  }
+
+  @Test
+  public void testNestedListVector1() throws Exception {
+    try (ListViewVector listViewVector = ListViewVector.empty("sourceVector", allocator)) {
+
+      MinorType listType = MinorType.LISTVIEW;
+      MinorType scalarType = MinorType.BIGINT;
+
+      listViewVector.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList1 = (ListViewVector) listViewVector.getDataVector();
+      innerList1.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList2 = (ListViewVector) innerList1.getDataVector();
+      innerList2.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList3 = (ListViewVector) innerList2.getDataVector();
+      innerList3.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList4 = (ListViewVector) innerList3.getDataVector();
+      innerList4.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList5 = (ListViewVector) innerList4.getDataVector();
+      innerList5.addOrGetVector(FieldType.nullable(listType.getType()));
+
+      ListViewVector innerList6 = (ListViewVector) innerList5.getDataVector();
+      innerList6.addOrGetVector(FieldType.nullable(scalarType.getType()));
+
+      listViewVector.setInitialCapacity(128);
+    }
+  }
+
```

Comment:
The following function will be served as an API, but just tested the idea before making a final decision on the API.

Reply:
Not valid anymore. Replaced this function with direct buffer setting.